src/Configurator/ModifyFileTrait.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Flex\Configurator;
+
+trait ModifyFileTrait
+{
+    protected function shouldWriteFile(string $file, bool $overwrite): bool
+    {
+        if (!file_exists($file)) {
+            return true;
+        }
+
+        if (!$overwrite) {
+            return false;
+        }
+
+        return
+            ($filePath = $this->getFilePathIfModified($file))
+            && !$this->io->askConfirmation("File $filePath exists and has modifications, overwrite?")
```

Comment:
`sprintf('File "%s" has uncommitted changes, overwrite?', $filePath)`
+ confirmation should default to `N`